server/storage/backend/batch_tx.go

```diff
@@ -105,7 +106,23 @@ func (t *batchTx) LockOutsideApply() {
 }
 
 func (t *batchTx) Unlock() {
-	if t.pending >= t.backend.batchLimit {
+	// We need to commit the transaction if there is any pending deleting
```

Comment:
I think this could make the purpose of this code clearer, my understanding is that:
* `t.pending >= t.backend.batchLimit` is for performance. We don't want buffer to grow too big as it has worse complexity.
* `t.pendingDeleteOperations > 0` is for correctness as buffer doesn't keep track of deletions.

Reply:
done